src/main/java/net/sf/jabref/logic/importer/fetcher/GvkFetcher.java

```diff
@@ -0,0 +1,135 @@
+        StringJoiner gvkQueryJoiner = new StringJoiner(" and ");
+        if (!searchKeys.contains(queryList.get(0))) {
+            throw new IllegalStateException("First element of query list has to be a searchKey");
+        };
+
+        Iterator<String> iterator = queryList.iterator();
+        String query = iterator.next();
+        // query is a searchKey
+        while (iterator.hasNext()) {
+            String searchKey = query;
+            StringJoiner parameterJoiner = new StringJoiner(" ");
+            while (iterator.hasNext() && (!searchKeys.contains((query = iterator.next())))) {
+                parameterJoiner.add(query);
+            }
+            gvkQueryJoiner.add("pica.".concat(searchKey).concat("=").concat(parameterJoiner.toString()));
+        }
+
+        return gvkQueryJoiner.toString();
+    }
+
+    String getSearchQueryString(String query) {
+        Objects.requireNonNull(query);
+        LinkedList<String> queryList = new LinkedList(Arrays.asList(query.split("\\s")));
+
+        if (queryList.isEmpty()) {
+            return "";
+        }
+
+        String gvkQuery;
+        if (searchKeys.contains(queryList.get(0))) {
+            gvkQuery = getSearchQueryStringForComplexQuery(queryList);
+        } else {
+            // query as pica.all
+            gvkQuery = queryList.stream().collect(Collectors.joining(" ", "pica.all=", ""));
+        }
+
+        return gvkQuery;
+    }
+
+    URL getQueryURL(String query) throws URISyntaxException, MalformedURLException {
+        String gvkQuery = getSearchQueryString(query);
+        final String URL_PATTERN = "http://sru.gbv.de/gvk?";
+        URIBuilder uriBuilder = new URIBuilder(URL_PATTERN);
+        uriBuilder.addParameter("version", "1.1");
+        uriBuilder.addParameter("operation", "searchRetrieve");
+        uriBuilder.addParameter("query", gvkQuery);
+        uriBuilder.addParameter("maximumRecords", "50");
+        uriBuilder.addParameter("recordSchema", "picaxml");
+        uriBuilder.addParameter("sortKeys", "Year,,1");
+        return uriBuilder.build().toURL();
+    }
+
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        if (query == null || query.trim().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<BibEntry> result;
+
+        try {
+            try (InputStream is = getQueryURL(query).openStream()) {
+                result = (new GVKParser()).parseEntries(is);
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("URI malformed error", e);
+            return Collections.emptyList();
+        } catch (IOException e) {
+            LOGGER.error("GVK: An I/O exception occurred", e);
+            return Collections.emptyList();
+        } catch (ParserConfigurationException e) {
+            LOGGER.error("GVK: An internal parser error occurred", e);
```

Comment:
Remove GVK prefix (information already included in class name used for the logger name)